tools/run_tests/sanity/sanity_tests.yaml

```diff
@@ -24,6 +24,9 @@
 - script: tools/distrib/check_pytype.sh
 - script: tools/distrib/clang_format_code.sh
 - script: tools/distrib/clang_tidy_code.sh
+  # ClangTidy needs to run exclusively because it runs bazel making changes
+  # and it needs many CPUs to parse all source files to be diagnosed.
+  cpu_cost: 1000
```

Comment:
I don't understand why running clang tidy as part of sanity check would need to modify anything (under sanity test, it should only try to clang tidy files in a separate subtree and then diff whether everything looks the same as the checked-in code). This should not collide with anything (same behavior as before).

Therefore I don't understand why we would need to run exclusively - that seems wrong, please explain.

Reply:
Let me retry this. It wasn't able to run deterministically before without this. I agree with you and it won't change anything except `compile_commands.json` but it didn't run successfully without this. Let me retry this.

Reply:
Oh I figured this out. This is still required because clangtidy needs files under the bazel output directory because the compilation database is built from bazel and it builds a new source directory view in that directory. (This is why it can access third_party files without having third_party sources in the repo) But this view can be removed or changed by another bazel invocation so it should run exclusively. I updated the comment.

Reply:
Running a couple of bazel at the same time would be fine. In this case, however, running bazel to build a compilation db and running clangtidy based on the db which have links to files in the bazel output directory. When another bazel build runs, it will change that directory while clangtidy is still working on it.